### PR TITLE
implementing UserInterface needs few more methods

### PR DESCRIPTION
**Pull request by jobayerccj**
Class must be declared abstract or implement methods 'eraseCredentials', 'getUsername', 'getPassword', 'getRoles', 'getSalt'

### CODE REVIEW

File: docs/cookbook/recipe_sonata_admin_without_user_bundle.rst [hunk 1]
```diff
@@ -26,38 +26,133 @@ User Entity
 This represents your security user, you can read more about it
 `here <https://symfony.com/doc/5.4/security.html#a-create-your-user-class>`__::
 
-    namespace App\Entity;
+    <?php
```
Comment (VincentLanglet) [inline docs/cookbook/recipe_sonata_admin_without_user_bundle.rst]: As you see the <?php tag wasn't needed and all the code was indented.
Please follow the same code style to have a successful build.

File: docs/cookbook/recipe_sonata_admin_without_user_bundle.rst [hunk 2]
```diff
@@ -26,38 +26,133 @@ User Entity
...
+    /**
+     * @see UserInterface
+     */
+    public function eraseCredentials()
+    {
+        // If you store any temporary, sensitive data on the user, clear it here
+        // $this->plainPassword = null;
     }
+}
+
```
Comment (VincentLanglet) [inline docs/cookbook/recipe_sonata_admin_without_user_bundle.rst]: There is an extra blank line

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: docs/cookbook/recipe_sonata_admin_without_user_bundle.rst (hunk 1 above)
Comment (jobayerccj) [reply docs/cookbook/recipe_sonata_admin_without_user_bundle.rst]: ok, thank you for your feedback

File: docs/cookbook/recipe_sonata_admin_without_user_bundle.rst [hunk 3]
```diff
@@ -26,38 +26,130 @@ User Entity
 This represents your security user, you can read more about it
 `here <https://symfony.com/doc/5.4/security.html#a-create-your-user-class>`__::
 
-    namespace App\Entity;
+namespace App\Entity;
```
Comment (VincentLanglet) [inline docs/cookbook/recipe_sonata_admin_without_user_bundle.rst]:
```suggestion
    namespace App\Entity;
```
Same for all the php code ^^

**Review by VincentLanglet — APPROVED** [review]

File: docs/cookbook/recipe_sonata_admin_without_user_bundle.rst [hunk 4]
```diff
@@ -28,35 +28,127 @@ This represents your security user, you can read more about it
...
     use Doctrine\ORM\Mapping as ORM;
+    use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
     use Symfony\Component\Security\Core\User\UserInterface;
 
     /**
-     * @ORM\Entity
+     * @ORM\Entity(repositoryClass=UserRepository::class)
      */
-    final class User implements UserInterface
+    class User implements UserInterface, PasswordAuthenticatedUserInterface
```
Comment (jordisala1991) [inline docs/cookbook/recipe_sonata_admin_without_user_bundle.rst:39]: I think we should mention that this interface is only available for sf > 4.4 (not sure which minor o 5 introduced it, I guess it was the 5.4)

Comment (jobayerccj) [reply docs/cookbook/recipe_sonata_admin_without_user_bundle.rst:39]: Yes, we can mention it.

**Review by jobayerccj — COMMENTED** [review]
checked & found ok

File: docs/cookbook/recipe_sonata_admin_without_user_bundle.rst (hunk 4 above)
Comment (jordisala1991) [reply docs/cookbook/recipe_sonata_admin_without_user_bundle.rst:39]: I dont see the mention in the docs, did you add it?

Comment (VincentLanglet) [reply docs/cookbook/recipe_sonata_admin_without_user_bundle.rst:39]: Introduce in 5.3, https://symfony.com/blog/new-in-symfony-5-3-improvements-for-security-users

@jobayerccj Can you add the mention ?

**Pull request closed without merging**